### PR TITLE
[csm] impl delete_data

### LINKED ISSUE #644: [CSM] add possibility to remove assigned data again.

### PR DESCRIPTION
**Pull request by marscher**
## Changes

Added method to remove assigned data from the graph.

## Related Issues

Closes #644 

## Checks

- [x] updated CHANGELOG.rst
- [x] updated tests
- [x] updated doc/
- [ ] update example/tutorial notebooks
- [ ] update manifest file

### CODE REVIEW

**Comment by marscher** [comment]
@vhirtham Now the data is also deleted for subsystems. Do I understand correctly, that subsystems are created exclusively by deleting and merging CSMs?

**Comment by vhirtham** [comment]
> @vhirtham Now the data is also deleted for subsystems. Do I understand correctly, that subsystems are created exclusively by deleting and merging CSMs?

Basically yes. The disassembly is also most relevant for the asdf serialization. In practice, we probably only use the merge function.

**Comment by CagtayFabry** [comment]
good to have this option 👍

File: weldx/transformations/cs_manager.py [hunk 1]
```diff
@@ -610,6 +610,35 @@ def assign_data(
...
+        ------
+        ValueError
+            If data_name has not been assigned yet.
+        """
+        # remove data from subsystems
+        for s in self._subsystems:
+            try:
+                s.data.remove(data_name)
+            except KeyError:
+                pass
```
Comment (vhirtham) [inline weldx/transformations/cs_manager.py:631]:
```suggestion
        for s in self._subsystems:
            s.data.discard(data_name)
```

**Review by vhirtham — APPROVED** [review]

File: weldx/transformations/cs_manager.py (hunk 1 above)
Comment (marscher) [reply weldx/transformations/cs_manager.py:631]: Didn't know about the discard method, what is the difference to remove?

Comment (marscher) [reply weldx/transformations/cs_manager.py:631]: Ah discard does not raise :)

Comment (vhirtham) [reply weldx/transformations/cs_manager.py]: jep, so we save some exception handling 😉

**Pull request merged**